### PR TITLE
[TS] LPS-155624

### PR DESCRIPTION
**Pull request by yuhai**
@tinatian, @dantewang, the pr is based on https://github.com/liferay-core-infra/liferay-portal/pull/1012 and our talked.

@ricardocousolr, I removed the integration test, instead of using unit test to expose the issue.

### CODE REVIEW

File: portal-impl/test/unit/com/liferay/portal/dao/sql/transformer/DB2SQLTransformerLogicTest.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ public void testReplaceCastText() {
 	public void testReplaceConcat() {
 		Assert.assertEquals(
 			"select * from Foo where foo LIKE CAST(bar AS VARCHAR(2000)) " +
-				"CONCAT COALESCE(CAST(? AS VARCHAR(2000)),'')",
+				"CONCAT ?",
```
Comment (dantewang) [inline portal-impl/test/unit/com/liferay/portal/dao/sql/transformer/DB2SQLTransformerLogicTest.java:69]: @yuhai Please change the commit message to something like "After reverting, the ? after CONCAT is not replaced. It's legal usage in DB2 whether the ? after CONCAT is replaced or not, so keep it untouched and adjust the test case".

Comment (yuhai) [reply portal-impl/test/unit/com/liferay/portal/dao/sql/transformer/DB2SQLTransformerLogicTest.java:69]: @dantewang, I just changed the commit message and pushed force it. Thanks.

**Review by dantewang — APPROVED** [review]

**Comment by tinatian** [comment]
@catarineaguiar Please help me scan test results.
@yuhai Code looks good to me, wait for QA to scan test results.

Thanks.

**Comment by tinatian** [comment]
ci:test:environments

**Comment by tinatian** [comment]
@shuyangzhou This pull is good to go.

Tina

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/120557

**Pull request closed without merging**